src/chksum.c

```diff
@@ -69,6 +76,72 @@ struct s_Chksum {
   } c;
 };
 
+#ifdef WITH_OPENSSL
+
+void
+openssl_ctx_copy(Chksum *chk_in, Chksum *chk_out)
+{
+  switch(chk_in->type)
+    {
+    case REPOKEY_TYPE_MD5:
+      solv_MD5_Init(&chk_out->c.md5);
+      EVP_MD_CTX_copy_ex(chk_in->c.md5, chk_out->c.md5);
```

Comment:
The code is weird. I read it and compared it with the OpenSSL documentation.
The documentation states:
`int EVP_MD_CTX_copy_ex(EVP_MD_CTX *out, const EVP_MD_CTX *in);`
Maybe the function arguments are swapped.

Also, I'm not sure if it is necessary to call `solv_SHA1_Init(&chk_out->c.md5)`. Just enough
`chk_out->c.md5 = EVP_MD_CTX_new();` ?

Reply:
Thanks for the feedback and well, you're definitely right. I somehow accidentally swapped the arguments even though I also read the man pages. The next weird thing is that I tested the code and it seemed cloning the context correctly as I've checked both have the same output when using `libdnf::repo::checksum_calc`. I'll fix that and try to look at this more carefully.

Regarding the initialization, the doc states that before using the `copy` function:

> out must be initialized before calling this function.

So I thought this is the purpose of the `*DigestInit*` functions. But I will try it only using the `_new()` and eventually I'll delete it.